addons/l10n_tw_edi_ecpay/models/account_move.py

```diff
@@ -0,0 +1,809 @@
+
+    def _reformat_phone_number(self, phone):
+        """
+        Cleans and reformats a phone number string by handling different input formats.
+
+        The method first replaces a leading plus sign ('+') with a '0' and removes any
+        following space. It then removes all non-digit characters (including spaces,
+        dashes, and parentheses) to return a clean, continuous number string.
+
+        :param phone: The phone number as a string.
+        :type phone: str
+        :return: A cleaned and reformatted phone number string.
+        :rtype: str
+
+        Example:
+            # Replaces leading '+' with '0' and removes spaces/dashes
+            _reformat_phone_number('+1 555-123-4567')  # returns '05551234567'
+
+            # Removes spaces and parentheses
+            _reformat_phone_number('(555) 123 4567')  # returns '5551234567'
+        """
+        cleaned_number = phone
+        # Replace leading '+' with '0'
+        if phone.startswith('+'):
+            if ' ' in phone:
+                parts = phone.split(' ', 1)
+                cleaned_number = '0' + parts[1]
+            else:
+                cleaned_number = '0' + phone[1:]
+        # Remove spaces, dashes, parentheses, etc.
+        cleaned_number = re.sub(r'[^\d+]', '', cleaned_number)
+        return cleaned_number
+
+    def _l10n_tw_edi_check_before_generate_invoice_json(self):
+        self.ensure_one()
+        errors = []
+        if not self.company_id.sudo().l10n_tw_edi_ecpay_merchant_id:
+            errors.append(self.env._("Please fill in the ECpay API information in the Setting!"))
+
+        if (self.l10n_tw_edi_is_print or self.partner_id.vat) and not self.partner_id.contact_address:
+            errors.append(self.env._("Please fill in the customer address for printing Ecpay invoice."))
+
+        if not self.partner_id.email and not self.partner_id.phone:
+            errors.append(self.env._("Please fill in the customer email or phone number for Ecpay invoice creation."))
+
+        if self.partner_id.phone:
+            if not self.partner_id.phone.startswith('+') or not re.match(r"\+\d{1,3} ", self.partner_id.phone):
+                errors.append(self.env._("Phone number must start with country code and please add a space after the country code (i.e +866 234 567 890)."))
+            formatted_phone = self._reformat_phone_number(self.partner_id.phone)
+            if not re.fullmatch(r'[\d+ ]+', formatted_phone):
+                errors.append(self.env._("Phone number contains invalid characters! Only digits, '+' and spaces are allowed."))
+
+        if self.l10n_tw_edi_is_b2b and not self.partner_id.vat:
+            errors.append(self.env._("Tax id is required for company contact or individual contact under a company."))
+
+        if self.l10n_tw_edi_is_b2b and self.partner_id.vat and (not self.partner_id.vat.isdigit() or len(self.partner_id.vat) != 8):
+            errors.append(self.env._("Tax id is invalid. Should be 8 digits."))
```

Comment:
```suggestion
            errors.append(self.env._("The tax ID is invalid. Should be 8 digits."))
```
Instead of saying "it should be 8 digits", maybe it would be better to give an example :thinking: 
```suggestion
            errors.append(self.env._("The tax ID is invalid. It should be in the format: '12345678'."))
```